### PR TITLE
Remove links to qiskit.org in API docs

### LINKED ISSUE #11289: Remove links to qiskit.org in API docs
qiskit.org is being sunset, per https://medium.com/qiskit/important-changes-to-qiskit-documentation-and-learning-resources-7f4e346b19ab. So, we should stop linking to `qiskit.org` in the API docs.

| Problem | Bad link | Redirect |
|--------|--------|--------|
| qiskit/utils/optionals.py 106 | https://qiskit.org/documentation/nature | https://qiskit.org/ecosystem/nature |
| qiskit/utils/mitigation/_filters.py 239 | https://qiskit.org/documentation/tutorials/noise/3_measurement_error_mitigation.html | delete the link since it's deprecated code |
| qiskit/qpy/__init__.py 483 | https://qiskit.org/documentation/tutorials/circuits_advanced/05_pulse_gates.html | https://docs.quantum-computing.ibm.com/build/pulse |
| qiskit/circuit/quantumcircuit.py 983 | qiskit.org/documentation/tutorials/circuits/3_summary_of_quantum_operations.html | https://docs.quantum-computing.ibm.com/build/circuit-construction | 
| qiskit/circuit/library/phase_estimation.py 47 | https://learn.qiskit.org/course/ch-algorithms/quantum-phase-estimation | https://github.com/Qiskit/textbook/blob/main/notebooks/ch-algorithms/quantum-phase-estimation.ipynb | 
| qiskit/qasm3/__init__.py 86 | https://qiskit.org/documentation/deprecation_policy.html | https://github.com/Qiskit/qiskit/blob/main/DEPRECATION.md | 
| qiskit/transpiler/passes/basis/basis_translator.py 205 | https://qiskit.org/documentation/stubs/qiskit.transpiler.passes. | Sphinx cross-reference |
| qiskit/transpiler/passes/basis/basis_translator.py 223 | https://qiskit.org/documentation/stubs/qiskit.transpiler.passes.BasisTranslator | Sphinx cross-reference |
| qiskit/transpiler/passes/scheduling/alignments/pulse_gate_validation.py 33 | https://qiskit.org/documentation/tutorials/circuits_advanced/05_pulse_gates.html | https://docs.quantum-computing.ibm.com/build/pulse |

### PR DESCRIPTION
**Pull request by gluonhiggs**
### Summary
fixes #11289 

### Details and comments
Removed links to qiskit.org in API docs::
    In ``qiskit/utils/optionals.py``,  	https://qiskit.org/documentation/nature is replaced with https://qiskit.org/ecosystem/nature
    In ``qiskit/utils/mitigation/_filters.py``, https://qiskit.org/documentation/tutorials/noise/3_measurement_error_mitigation.html is deleted
    In ``qiskit/qpy/init.py``, https://qiskit.org/documentation/tutorials/circuits_advanced/05_pulse_gates.html is replaced
    with https://docs.quantum-computing.ibm.com/build/pulse
    In ``qiskit/circuit/quantumcircuit.py``, https://qiskit.org/documentation/tutorials/circuits/3_summary_of_quantum_operations.html
    is replaced with https://docs.quantum-computing.ibm.com/build/circuit-construction
    In ``qiskit/circuit/library/phase_estimation.py``, https://learn.qiskit.org/course/ch-algorithms/quantum-phase-estimation is
    replaced with https://github.com/Qiskit/textbook/blob/main/notebooks/ch-algorithms/quantum-phase-estimation.ipynb
    In ``qiskit/qasm3/init.py``, https://qiskit.org/documentation/deprecation_policy.html is replaced with
    https://github.com/Qiskit/qiskit/blob/main/DEPRECATION.md
    In ``qiskit/transpiler/passes/basis/basis_translator.py``, https://qiskit.org/documentation/stubs/qiskit.transpiler.passes.BasisTranslator 
    is replaced with Sphinx cross-reference
    In ``qiskit/transpiler/passes/scheduling/alignments/pulse_gate_validation.py``, https://qiskit.org/documentation/tutorials/circuits_advanced/05_pulse_gates.html
    is replaced with https://docs.quantum-computing.ibm.com/build/pulse

### CODE REVIEW

File: qiskit/transpiler/passes/basis/basis_translator.py [hunk 1]
```diff
@@ -202,8 +202,7 @@ def run(self, dag):
                     "target basis is not universal or there are additional equivalence rules "
                     "needed in the EquivalenceLibrary being used. For more details on this "
                     "error see: "
-                    "https://qiskit.org/documentation/stubs/qiskit.transpiler.passes."
-                    "BasisTranslator.html#translation_errors"
+                    "Sphinx cross-reference"
```
Comment (ElePT) [inline qiskit/transpiler/passes/basis/basis_translator.py]: Sphinx cross-references are links to other parts of the docs, you can check out the  [sphinx documentation](https://docs.readthedocs.io/en/stable/guides/cross-referencing-with-sphinx.html) to see different strategies to incorporate cross-references into the text. In this case, you would want to link to the "translation_errors" section of the `BasisTranslator` docstring. I think that there is already a tag you can use on line [86](https://github.com/Qiskit/qiskit/blob/915195ee27ade2ba60075b854460e2047dd68939/qiskit/transpiler/passes/basis/basis_translator.py#L86C1-L86C28), so the cross-reference link should work if you do (check out the explicit reference section in the sphinx docs above):

```suggestion
                    `translation_errors`_
```

File: qiskit/transpiler/passes/basis/basis_translator.py [hunk 2]
```diff
@@ -220,8 +219,7 @@ def run(self, dag):
                 f"basis: {list(target_basis)}. This likely means the target basis is not universal "
                 "or there are additional equivalence rules needed in the EquivalenceLibrary being "
                 "used. For more details on this error see: "
-                "https://qiskit.org/documentation/stubs/qiskit.transpiler.passes.BasisTranslator."
-                "html#translation_errors"
+                "Sphinx cross-reference"
```
Comment (ElePT) [inline qiskit/transpiler/passes/basis/basis_translator.py]: Same comment as above :) I think you can also move the cross reference to the line above.

**Review by ElePT — COMMENTED** [review]
Thank you for picking up this issue so quickly! Most of the changes look good to me, I only have 2 comments:
1. The sphinx cross references are missing (see the comment where I try to explain what they are and how to do them)
2. Docs changes are not included in the changelog, so the correct action in this case is not to add a release note. Thanks though for going through the effort of making it :)

Once these comments are addressed, the PR should be ready to go.

File: qiskit/transpiler/passes/basis/basis_translator.py (hunk 2 above)
Comment (gluonhiggs) [reply qiskit/transpiler/passes/basis/basis_translator.py]: Please have a look! I added a commit as you suggested.

File: qiskit/transpiler/passes/basis/basis_translator.py [hunk 3]
```diff
@@ -219,9 +217,7 @@ def run(self, dag):
                 f"{[x[0] for x in source_basis]} to the backend's (or manually specified) target "
                 f"basis: {list(target_basis)}. This likely means the target basis is not universal "
                 "or there are additional equivalence rules needed in the EquivalenceLibrary being "
-                "used. For more details on this error see: "
-                "https://qiskit.org/documentation/stubs/qiskit.transpiler.passes.BasisTranslator."
-                "html#translation_errors"
+                "used. For more details on this error see: `translation_errors`_"
```
Comment (ElePT) [inline qiskit/transpiler/passes/basis/basis_translator.py]:
```suggestion
                "used. For more details on this error see: "
                "https://docs.quantum-computing.ibm.com/api/qiskit/qiskit.transpiler.passes."
                "BasisTranslator#translation-errors"
```

File: qiskit/transpiler/passes/basis/basis_translator.py [hunk 4]
```diff
@@ -201,9 +201,7 @@ def run(self, dag):
                     f"specified) target basis: {list(expanded_target)}. This likely means the "
                     "target basis is not universal or there are additional equivalence rules "
                     "needed in the EquivalenceLibrary being used. For more details on this "
-                    "error see: "
-                    "https://qiskit.org/documentation/stubs/qiskit.transpiler.passes."
-                    "BasisTranslator.html#translation_errors"
+                    "error see: `translation_errors`_"
```
Comment (ElePT) [inline qiskit/transpiler/passes/basis/basis_translator.py]:
```suggestion
```suggestion
                "error see: "
                "https://docs.quantum-computing.ibm.com/api/qiskit/qiskit.transpiler.passes."
                "BasisTranslator#translation-errors"
```
```

**Review by ElePT — COMMENTED** [review]
Thanks for applying my suggestion. I took a closer look at the built documentation to make sure the links worked, and realized that the sphinx cross references we discussed were not showing up. Then I expanded the code and realized that they are in `TranspileError` messages, which display plain text, and are not parts of the docs per se. I am reverting my suggestion and suggesting a new link instead, as the sphinx references will not work. This was probably an oversight in the initial analysis of the issue.

File: qiskit/transpiler/passes/basis/basis_translator.py (hunk 4 above)
Comment (gluonhiggs) [reply qiskit/transpiler/passes/basis/basis_translator.py]: I checked the link and added a new commit!

**Review by ElePT — APPROVED** [review]
LGTM! Thanks a lot, there was a merge conflict because of a PR merged a few minutes ago but I have handled it myself.

**Comment by Eric-Arellano** [comment]
Thanks so much @gluonhiggs! And thanks @ElePT for helping with the review :)

**Pull request merged**